File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    type Dictionary = {
        [key:number]:number;
    }

    let myDictionary: Dictionary = {};
    
    for (let i = 0; i < nums.length; i++) {
        let complement = target - nums[i]
        if (complement in myDictionary) {
            return [i, myDictionary[complement]];
        } else {
            myDictionary[nums[i]] = i;
        }
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        visited = {}
        for index, n in enumerate(nums):
            compliment = target - n
            if compliment in visited:
                return [index, visited[compliment]]
            else:
                visited[n] = index